packages/qwik-city/runtime/src/image-component.tsx

```diff
@@ -0,0 +1,227 @@
+      };
+      return {
+        ...baseStyles,
+        ...heightStyle,
+        width: '100%',
+        'aspect-ratio': isValid(aspectRatio) ? `${aspectRatio}` : undefined,
+      };
+    }
+  }
+};
+
+export const getSizes = ({ width, layout }: Pick<ImageProps, 'width' | 'layout'>) => {
+  if (!width || !layout) {
+    return undefined;
+  }
+  switch (layout) {
+    case `constrained`:
+      return `(min-width: ${width}px) ${width}px, 100vw`;
+    case `fixed`:
+      return `${width}px`;
+    case `fullWidth`:
+      return `100vw`;
+
+    default:
+      return undefined;
+  }
+};
+
+export const getSrcSet = async ({
+  src = '',
+  width,
+  height,
+  aspectRatio,
+  layout,
+  resolutions,
+  imageTransformer$,
+}: Pick<ImageProps, 'src' | 'width' | 'height' | 'aspectRatio' | 'layout'> &
+  ImageState): Promise<string> => {
+  const breakpoints = getBreakpoints({
+    width: typeof width === 'string' ? parseInt(width, 10) : width,
+    layout,
+    resolutions: resolutions || DEFAULT_RESOLUTIONS,
+  });
+
+  const srcSets = [];
+  for await (const breakpoint of breakpoints.sort()) {
+    let transformedHeight = typeof height === 'string' ? parseInt(height, 10) : height;
+    if (height && aspectRatio) {
+      transformedHeight = Math.round(breakpoint * aspectRatio);
+    }
+
+    if (!imageTransformer$) {
+      srcSets.push(`${src} ${breakpoint}w`);
+      continue;
+    }
+
+    const transformed = await imageTransformer$({
+      src,
+      width: breakpoint,
+      height: transformedHeight,
+    });
+
+    srcSets.push(`${transformed} ${breakpoint}w`);
+  }
+
+  return srcSets.join(',\n');
+};
+
+export const getBreakpoints = ({
+  width: widthAttribute,
+  layout,
+  resolutions = [],
+}: Pick<ImageProps, 'width' | 'layout'> & Pick<ImageState, 'resolutions'>): number[] => {
+  if (layout === 'fullWidth') {
+    return resolutions;
+  }
+  if (!widthAttribute) {
+    return [];
+  }
+  const width = typeof widthAttribute === 'string' ? parseInt(widthAttribute, 10) : widthAttribute;
+  const doubleWidth = width * 2;
+  if (layout === 'fixed') {
+    return [width, doubleWidth];
+  }
+  if (layout === 'constrained') {
+    return [width, doubleWidth, ...resolutions.filter((w) => w < doubleWidth)];
+  }
+
+  return [];
+};
+
+/**
+ * @alpha
+ */
+export const Image = component$<ImageProps>((props) => {
+  const state = useContext(ImageContext);
+  const { resolutions, imageTransformer$, ...imageAttributes } = {
+    ...state,
+    ...props,
+  };
+  const style = { ...props.style, ...getStyles(props) };
+  const sizes = getSizes(props);
+  const srcSetSignal = useSignal('');
+
+  const { src, width, height, aspectRatio, layout } = props;
+  useTask$(async () => {
+    srcSetSignal.value = await getSrcSet({
+      src,
+      width,
+      height,
+      aspectRatio,
+      layout,
+      resolutions,
+      imageTransformer$,
+    });
+  });
+
+  return (
+    <img
+      id={useId()}
+      decoding="async"
```

Comment:
what do you think about defining a default for `loading` as well? setting it to `lazy` ? for an LCP relevant image you would most likely go with `decoding="sync" loading="eager"`, right?

Reply:
Yes, let's add loading="lazy" by default

Reply:
done 🚀